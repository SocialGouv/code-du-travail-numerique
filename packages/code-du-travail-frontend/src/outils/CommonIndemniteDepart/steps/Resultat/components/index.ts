export { default as ForMoreInfo } from "./ForMoreInfo";
export { default as DecryptResult } from "./DecryptResult";
export { default as FormulaInterpreter } from "./FormulaInterpreter";
export { default as FilledElements } from "./FilledElements";
export { default as Result } from "./Result";
export { default as ErrorPublicodes } from "./ErrorPublicodes";
