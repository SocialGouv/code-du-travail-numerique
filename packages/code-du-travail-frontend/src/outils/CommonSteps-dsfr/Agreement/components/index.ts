export { default as AgreementSearch } from "./AgreementSearch";
export { default as EnterpriseSearch } from "./EnterpriseSearch";
export * from "./NoEnterprise";
