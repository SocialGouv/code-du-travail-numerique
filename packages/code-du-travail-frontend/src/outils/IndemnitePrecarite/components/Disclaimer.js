import { icons, IconStripe } from "@socialgouv/cdtn-ui";
import React from "react";

import {
  Warning,
  WarningTitle,
} from "../../DureePreavisRetraite/steps/component/WarningResult";

function Disclaimer({ situation, idcc }) {
  if (idcc > 0 && situation.idcc > 0) {
    return (
      <Warning>
        <IconStripe centered icon={icons.Warning}>
          <WarningTitle>
            Attention il peut exister un montant plus favorable
          </WarningTitle>
        </IconStripe>
        <p>
          Un accord d’entreprise peut prévoir un montant différent qu’il soit
          plus élevé ou plus faible. Dans ce cas, s’applique le montant prévu
          par l’accord d’entreprise, sauf si le contrat de travail prévoit un
          montant plus favorable pour le salarié.
        </p>
        {!situation.hasConventionalProvision && (
          <p>
            Attention, dans le cas où l’accord d’entreprise prévoit un taux
            inférieur à 10% dans la limite de 6%, il doit y avoir des
            contreparties offertes au salarié, notamment sous la forme d’un
            accès privilégié à la formation professionnelle (action de
            formation, bilan de compétences).
          </p>
        )}
      </Warning>
    );
  }
  // case for no ccn provided or unhandled ccn
  return (
    <Warning>
      <IconStripe centered icon={icons.Warning}>
        <WarningTitle>
          Attention il peut exister un montant plus favorable
        </WarningTitle>
      </IconStripe>
      <p>
        Une convention collective de branche étendue ou un accord d’entreprise
        peut prévoir un montant différent qu’il soit plus élevé ou plus faible
        que celui prévu par le code du travail.
      </p>
      <p>
        Attention, dans le cas où la convention ou l’accord collectif prévoit un
        taux inférieur à 10% dans la limite de 6%, il doit y avoir des
        contreparties offertes au salarié, notamment sous la forme d’un accès
        privilégié à la formation professionnelle (action de formation, bilan de
        compétences). Dans tous les cas, le contrat de travail peut prévoir un
        montant plus favorable pour le salarié. Il faut alors appliquer ce
        montant.
      </p>
    </Warning>
  );
}

export default Disclaimer;
