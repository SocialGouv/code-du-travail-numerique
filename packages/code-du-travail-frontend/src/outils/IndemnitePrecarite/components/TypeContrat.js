import { InputRadio } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React from "react";
import { Field } from "react-final-form";

import { ErrorField } from "../../common/ErrorField";
import { Question } from "../../common/Question";
import { required } from "../../common/validators";

export const CONTRACT_TYPE = {
  CDD: "CDD",
  CTT: "CTT",
};

function TypeContrat({ name, onChange }) {
  return (
    <>
      <Question required>
        Quel est le type du contrat de travail&nbsp;?
      </Question>
      <Field
        type="radio"
        name={name}
        value={CONTRACT_TYPE.CDD}
        id={CONTRACT_TYPE.CDD}
        validate={required}
      >
        {(props) => (
          <InputRadio
            id={`${name}-cdd`}
            label="Contrat à durée déterminée (CDD)"
            {...props.input}
            onChange={(...changeProps) => {
              onChange(...changeProps);
              props.input.onChange(...changeProps);
            }}
          />
        )}
      </Field>
      <Field
        type="radio"
        name={name}
        value={CONTRACT_TYPE.CTT}
        id={CONTRACT_TYPE.CTT}
        validate={required}
      >
        {(props) => (
          <InputRadio
            id={`${name}-ci`}
            label="Contrat de travail temporaire (Contrat d’intérim)"
            {...props.input}
            onChange={(...changeProps) => {
              onChange(...changeProps);
              props.input.onChange(...changeProps);
            }}
          />
        )}
      </Field>
      <ErrorField name={name} />
    </>
  );
}

TypeContrat.propTypes = {
  name: PropTypes.string.isRequired,
};

export { TypeContrat };
