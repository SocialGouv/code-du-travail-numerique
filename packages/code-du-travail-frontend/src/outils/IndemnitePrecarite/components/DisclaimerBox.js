import React from "react";

import Disclaimer from "../../common/Disclaimer";

function DisclaimerBox({ situation, idcc }) {
  if (idcc > 0 && situation.idcc > 0) {
    return (
      <Disclaimer title={"Attention il peut exister un montant plus favorable"}>
        <p>
          Un accord d’entreprise peut prévoir un montant différent qu’il soit
          plus élevé ou plus faible. Dans ce cas, s’applique le montant prévu
          par l’accord d’entreprise, sauf si le contrat de travail prévoit un
          montant plus favorable pour le salarié.
        </p>
        {!situation.hasConventionalProvision && (
          <p>
            Attention, dans le cas où l’accord d’entreprise prévoit un taux
            inférieur à 10% dans la limite de 6%, il doit y avoir des
            contreparties offertes au salarié, notamment sous la forme d’un
            accès privilégié à la formation professionnelle (action de
            formation, bilan de compétences).
          </p>
        )}
      </Disclaimer>
    );
  }
  // case for no ccn provided or unhandled ccn
  return (
    <Disclaimer title={"Attention il peut exister un montant plus favorable"}>
      <p>
        Une convention collective de branche étendue ou un accord d’entreprise
        peut prévoir un montant différent qu’il soit plus élevé ou plus faible
        que celui prévu par le code du travail.
      </p>
      <p>
        Attention, dans le cas où la convention ou l’accord collectif prévoit un
        taux inférieur à 10% dans la limite de 6%, il doit y avoir des
        contreparties offertes au salarié, notamment sous la forme d’un accès
        privilégié à la formation professionnelle (action de formation, bilan de
        compétences). Dans tous les cas, le contrat de travail peut prévoir un
        montant plus favorable pour le salarié. Il faut alors appliquer ce
        montant.
      </p>
    </Disclaimer>
  );
}

export default DisclaimerBox;
