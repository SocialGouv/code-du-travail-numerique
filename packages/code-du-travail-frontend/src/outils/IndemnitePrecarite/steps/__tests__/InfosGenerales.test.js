import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { Form } from "react-final-form";

import { CONTRACT_TYPE } from "../../components/TypeContrat";
import { StepInfosGenerales } from "../InfosGenerales";

function renderForm(data) {
  return render(
    <Form
      validate={StepInfosGenerales.validate}
      initialValues={{ ...data }}
      onSubmit={jest.fn()}
    >
      {({ handleSubmit, form }) => (
        <form onSubmit={handleSubmit}>
          <StepInfosGenerales form={form} />
          <button data-testid="nextBt">suivant</button>
        </form>
      )}
    </Form>
  );
}

describe("<StepInfosGenerales />", () => {
  it("should render", () => {
    const { container } = renderForm({});
    expect(container).toMatchSnapshot();
  });

  it("should render with CDD", () => {
    const { container } = renderForm({
      contractType: CONTRACT_TYPE.CDD,
    });
    expect(container).toMatchSnapshot();
  });

  it("should render with CDD and not handled ccn", () => {
    const { container } = renderForm({
      ccn: { convention: { num: 16 } },
      contractType: CONTRACT_TYPE.CDD,
    });
    expect(container).toMatchSnapshot();
  });

  it("should render with CTT", () => {
    const { container } = renderForm({ contractType: CONTRACT_TYPE.CTT });
    expect(container).toMatchSnapshot();
  });

  it("should display and error if ctt and contrat is mission-formation", () => {
    const { getByTestId, getByText } = renderForm({
      contractType: CONTRACT_TYPE.CTT,
    });
    const missionFormation =
      getByTestId("cttFormation").querySelector(`input[value=true]`);
    fireEvent.click(missionFormation);
    // validate form
    const validateButton = getByTestId("nextBt");
    validateButton.click();
    expect(getByText(/ne permet pas/i)).toBeTruthy();
  });
  it("should display and info alert if ctt and contrat is not mission-formation", () => {
    const { getByTestId, getByText } = renderForm({
      contractType: CONTRACT_TYPE.CTT,
    });
    const missionFormation =
      getByTestId("cttFormation").querySelector(`input[value=false]`);
    fireEvent.click(missionFormation);
    // validate form
    const validateButton = getByTestId("nextBt");
    validateButton.click();
    expect(getByText(/attention/i)).toBeTruthy();
  });
});
