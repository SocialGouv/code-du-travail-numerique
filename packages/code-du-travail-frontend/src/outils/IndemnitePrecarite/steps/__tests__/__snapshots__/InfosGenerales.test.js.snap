// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInfosGenerales /> should render 1`] = `
.c1 {
  color: #3e486e;
  font-size: 1.8rem;
  font-weight: 600;
}

.c2 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: calc(1.6rem * 1.25);
  height: calc(1.6rem * 1.25);
  margin: calc(1.6rem / 5) calc(2 * 1.6rem / 3) 0 0;
  padding: 0;
  line-height: inherit;
  background: #fff;
  border: 1px solid #bbcadf;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:before {
  position: absolute;
  top: 50%;
  left: 50%;
  display: none;
  width: calc(1.6rem / 1.5);
  height: calc(1.6rem / 1.5);
  background-color: #f66663;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  content: "";
}

.c4:checked {
  background-color: #fff;
  border-color: #f66663;
}

.c4:checked:before {
  display: block;
}

.c0 {
  display: block;
  margin: 1rem 0;
  cursor: pointer;
}

@media (max-width:600px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<div>
  <form>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        Quel est le type du contrat de travail ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c3"
    >
      <input
        class="c4"
        id="contractType-cdd"
        name="contractType"
        type="radio"
        value="CDD"
      />
      <label
        class="c5"
        for="contractType-cdd"
      >
        Contrat à durée déterminée (CDD)
      </label>
    </div>
    <div
      class="c3"
    >
      <input
        class="c4"
        id="contractType-ci"
        name="contractType"
        type="radio"
        value="CTT"
      />
      <label
        class="c5"
        for="contractType-ci"
      >
        Contrat de travail temporaire (Contrat d’intérim)
      </label>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;

exports[`<StepInfosGenerales /> should render with CDD 1`] = `
.c1 {
  color: #3e486e;
  font-size: 1.8rem;
  font-weight: 600;
}

.c2 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: calc(1.6rem * 1.25);
  height: calc(1.6rem * 1.25);
  margin: calc(1.6rem / 5) calc(2 * 1.6rem / 3) 0 0;
  padding: 0;
  line-height: inherit;
  background: #fff;
  border: 1px solid #bbcadf;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:before {
  position: absolute;
  top: 50%;
  left: 50%;
  display: none;
  width: calc(1.6rem / 1.5);
  height: calc(1.6rem / 1.5);
  background-color: #f66663;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  content: "";
}

.c4:checked {
  background-color: #fff;
  border-color: #f66663;
}

.c4:checked:before {
  display: block;
}

.c7 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
}

.c10 {
  position: absolute;
  top: 1.6rem;
  right: 1.6rem;
  width: 1.6rem;
  height: 1.6rem;
  color: #f66663;
  pointer-events: none;
}

.c9 {
  width: 100%;
  height: 5.4rem;
  padding: 0 2rem 0;
  padding-right: 5rem;
  color: #3e486e;
  font-size: 1.6rem;
  font-family: "Open Sans",sans-serif;
  vertical-align: middle;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: border-color 250ms ease;
  transition: border-color 250ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c9::-ms-expand {
  background-color: transparent;
  border: 0 transparent;
}

.c9 *::-ms-backdrop,
.c9 {
  padding-right: 1.6rem;
}

.c9:invalid {
  border-color: #eb5757;
}

.c9:disabled {
  background-color: #e4e8ef;
}

.c0 {
  display: block;
  margin: 1rem 0;
  cursor: pointer;
}

.c8 {
  width: 40rem;
}

.c6 {
  margin-bottom: 1.6rem;
}

@media (max-width:600px) {
  .c7 {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c0 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <form>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        Quel est le type du contrat de travail ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c3"
    >
      <input
        checked=""
        class="c4"
        id="contractType-cdd"
        name="contractType"
        type="radio"
        value="CDD"
      />
      <label
        class="c5"
        for="contractType-cdd"
      >
        Contrat à durée déterminée (CDD)
      </label>
    </div>
    <div
      class="c3"
    >
      <input
        class="c4"
        id="contractType-ci"
        name="contractType"
        type="radio"
        value="CTT"
      />
      <label
        class="c5"
        for="contractType-ci"
      >
        Contrat de travail temporaire (Contrat d’intérim)
      </label>
    </div>
    <div
      class="c6"
    >
      <label
        class="c0"
        for="input-criteria.cddType"
      >
        <span
          class="c1"
        >
          Quel est le type de CDD ?
        </span>
        <span
          class="c2"
        >
           (obligatoire)
        </span>
      </label>
      <div
        class="c7 c8"
      >
        <select
          class="c9"
          data-testid="criteria.cddType"
          id="input-criteria.cddType"
          name="criteria.cddType"
        >
          <option
            disabled=""
            value=""
          >
            ...
          </option>
          <option
            value="CDD saisonnier"
          >
            CDD saisonnier
          </option>
          <option
            value="CDD d'usage"
          >
            CDD d'usage
          </option>
          <option
            value="CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires"
          >
            CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires
          </option>
          <option
            value="CCD dans le cadre d’un congé de mobilité"
          >
            CCD dans le cadre d’un congé de mobilité
          </option>
          <option
            value="Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)"
          >
            Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)
          </option>
          <option
            value="Contrat de professionnalisation ou Contrat d’apprentissage"
          >
            Contrat de professionnalisation ou Contrat d’apprentissage
          </option>
          <option
            value="Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié"
          >
            Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié
          </option>
          <option
            value="Autres"
          >
            Autres
          </option>
        </select>
        <div
          aria-hidden="true"
          class="c10"
        >
          <svg
            aria-hidden="true"
            fill="none"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#arrow-down_svg__a)"
            >
              <path
                d="m8.425 11.573 7.11-6.072a.709.709 0 0 0 .26-.537.709.709 0 0 0-.26-.537.973.973 0 0 0-.629-.223.973.973 0 0 0-.63.223l-6.48 5.536-6.481-5.536a.973.973 0 0 0-.63-.223.973.973 0 0 0-.628.223.708.708 0 0 0-.261.537c0 .202.094.395.26.537l7.11 6.072a.91.91 0 0 0 .29.165 1.021 1.021 0 0 0 .68 0 .91.91 0 0 0 .29-.165z"
                fill="currentColor"
              />
            </g>
            <defs>
              <clippath
                id="arrow-down_svg__a"
              >
                <path
                  d="M0 0h16v16H0z"
                  fill="currentColor"
                />
              </clippath>
            </defs>
          </svg>
        </div>
      </div>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;

exports[`<StepInfosGenerales /> should render with CDD and not handled ccn 1`] = `
.c1 {
  color: #3e486e;
  font-size: 1.8rem;
  font-weight: 600;
}

.c2 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: calc(1.6rem * 1.25);
  height: calc(1.6rem * 1.25);
  margin: calc(1.6rem / 5) calc(2 * 1.6rem / 3) 0 0;
  padding: 0;
  line-height: inherit;
  background: #fff;
  border: 1px solid #bbcadf;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:before {
  position: absolute;
  top: 50%;
  left: 50%;
  display: none;
  width: calc(1.6rem / 1.5);
  height: calc(1.6rem / 1.5);
  background-color: #f66663;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  content: "";
}

.c4:checked {
  background-color: #fff;
  border-color: #f66663;
}

.c4:checked:before {
  display: block;
}

.c7 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
}

.c10 {
  position: absolute;
  top: 1.6rem;
  right: 1.6rem;
  width: 1.6rem;
  height: 1.6rem;
  color: #f66663;
  pointer-events: none;
}

.c9 {
  width: 100%;
  height: 5.4rem;
  padding: 0 2rem 0;
  padding-right: 5rem;
  color: #3e486e;
  font-size: 1.6rem;
  font-family: "Open Sans",sans-serif;
  vertical-align: middle;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: border-color 250ms ease;
  transition: border-color 250ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c9::-ms-expand {
  background-color: transparent;
  border: 0 transparent;
}

.c9 *::-ms-backdrop,
.c9 {
  padding-right: 1.6rem;
}

.c9:invalid {
  border-color: #eb5757;
}

.c9:disabled {
  background-color: #e4e8ef;
}

.c0 {
  display: block;
  margin: 1rem 0;
  cursor: pointer;
}

.c8 {
  width: 40rem;
}

.c6 {
  margin-bottom: 1.6rem;
}

@media (max-width:600px) {
  .c7 {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c0 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <form>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        Quel est le type du contrat de travail ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c3"
    >
      <input
        checked=""
        class="c4"
        id="contractType-cdd"
        name="contractType"
        type="radio"
        value="CDD"
      />
      <label
        class="c5"
        for="contractType-cdd"
      >
        Contrat à durée déterminée (CDD)
      </label>
    </div>
    <div
      class="c3"
    >
      <input
        class="c4"
        id="contractType-ci"
        name="contractType"
        type="radio"
        value="CTT"
      />
      <label
        class="c5"
        for="contractType-ci"
      >
        Contrat de travail temporaire (Contrat d’intérim)
      </label>
    </div>
    <div
      class="c6"
    >
      <label
        class="c0"
        for="input-criteria.cddType"
      >
        <span
          class="c1"
        >
          Quel est le type de CDD ?
        </span>
        <span
          class="c2"
        >
           (obligatoire)
        </span>
      </label>
      <div
        class="c7 c8"
      >
        <select
          class="c9"
          data-testid="criteria.cddType"
          id="input-criteria.cddType"
          name="criteria.cddType"
        >
          <option
            disabled=""
            value=""
          >
            ...
          </option>
          <option
            value="CDD saisonnier"
          >
            CDD saisonnier
          </option>
          <option
            value="CDD d'usage"
          >
            CDD d'usage
          </option>
          <option
            value="CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires"
          >
            CDD conclu avec un jeune (mineur ou majeur) pendant ses vacances scolaires ou universitaires
          </option>
          <option
            value="CCD dans le cadre d’un congé de mobilité"
          >
            CCD dans le cadre d’un congé de mobilité
          </option>
          <option
            value="Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)"
          >
            Contrat unique d’insertion (CUI) ou Parcours emploi compétences (PEC)
          </option>
          <option
            value="Contrat de professionnalisation ou Contrat d’apprentissage"
          >
            Contrat de professionnalisation ou Contrat d’apprentissage
          </option>
          <option
            value="Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié"
          >
            Contrat pour lequel l’employeur s’est engagé à assurer un complément de formation professionnelle au salarié
          </option>
          <option
            value="Autres"
          >
            Autres
          </option>
        </select>
        <div
          aria-hidden="true"
          class="c10"
        >
          <svg
            aria-hidden="true"
            fill="none"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#arrow-down_svg__a)"
            >
              <path
                d="m8.425 11.573 7.11-6.072a.709.709 0 0 0 .26-.537.709.709 0 0 0-.26-.537.973.973 0 0 0-.629-.223.973.973 0 0 0-.63.223l-6.48 5.536-6.481-5.536a.973.973 0 0 0-.63-.223.973.973 0 0 0-.628.223.708.708 0 0 0-.261.537c0 .202.094.395.26.537l7.11 6.072a.91.91 0 0 0 .29.165 1.021 1.021 0 0 0 .68 0 .91.91 0 0 0 .29-.165z"
                fill="currentColor"
              />
            </g>
            <defs>
              <clippath
                id="arrow-down_svg__a"
              >
                <path
                  d="M0 0h16v16H0z"
                  fill="currentColor"
                />
              </clippath>
            </defs>
          </svg>
        </div>
      </div>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;

exports[`<StepInfosGenerales /> should render with CTT 1`] = `
.c1 {
  color: #3e486e;
  font-size: 1.8rem;
  font-weight: 600;
}

.c2 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: calc(1.6rem * 1.25);
  height: calc(1.6rem * 1.25);
  margin: calc(1.6rem / 5) calc(2 * 1.6rem / 3) 0 0;
  padding: 0;
  line-height: inherit;
  background: #fff;
  border: 1px solid #bbcadf;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:before {
  position: absolute;
  top: 50%;
  left: 50%;
  display: none;
  width: calc(1.6rem / 1.5);
  height: calc(1.6rem / 1.5);
  background-color: #f66663;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  content: "";
}

.c4:checked {
  background-color: #fff;
  border-color: #f66663;
}

.c4:checked:before {
  display: block;
}

.c0 {
  display: block;
  margin: 1rem 0;
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 1.6rem;
}

@media (max-width:600px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<div>
  <form>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        Quel est le type du contrat de travail ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c3"
    >
      <input
        class="c4"
        id="contractType-cdd"
        name="contractType"
        type="radio"
        value="CDD"
      />
      <label
        class="c5"
        for="contractType-cdd"
      >
        Contrat à durée déterminée (CDD)
      </label>
    </div>
    <div
      class="c3"
    >
      <input
        checked=""
        class="c4"
        id="contractType-ci"
        name="contractType"
        type="radio"
        value="CTT"
      />
      <label
        class="c5"
        for="contractType-ci"
      >
        Contrat de travail temporaire (Contrat d’intérim)
      </label>
    </div>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        S’agit-il d’un contrat de mission-formation ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c6"
      data-testid="cttFormation"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="cttFormation-oui"
          name="cttFormation"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="cttFormation-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="cttFormation-non"
          name="cttFormation"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="cttFormation-non"
        >
          Non
        </label>
      </div>
    </div>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        Le contrat d'intérim a-t-il été rompu avant la fin prévue pour une des raisons suivantes : la propre initiative du salarié, la faute grave du salarié, cas de force majeure ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c6"
      data-testid="ruptureContratFauteGrave"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="ruptureContratFauteGrave-oui"
          name="ruptureContratFauteGrave"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="ruptureContratFauteGrave-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="ruptureContratFauteGrave-non"
          name="ruptureContratFauteGrave"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="ruptureContratFauteGrave-non"
        >
          Non
        </label>
      </div>
    </div>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        À la fin du contrat d'intérim, le salarié a-t-il été immédiatement embauché en CDI au sein de l'entreprise dans laquelle il effectuait sa mission ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c6"
      data-testid="propositionCDIFinContrat"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="propositionCDIFinContrat-oui"
          name="propositionCDIFinContrat"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="propositionCDIFinContrat-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="propositionCDIFinContrat-non"
          name="propositionCDIFinContrat"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="propositionCDIFinContrat-non"
        >
          Non
        </label>
      </div>
    </div>
    <label
      class="c0"
    >
      <span
        class="c1"
      >
        Le salarié a-t-il refusé la mise en œuvre de la souplesse prévue dans le contrat d’intérim ?
      </span>
      <span
        class="c2"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c6"
      data-testid="refusSouplesse"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="refusSouplesse-oui"
          name="refusSouplesse"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="refusSouplesse-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="refusSouplesse-non"
          name="refusSouplesse"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="refusSouplesse-non"
        >
          Non
        </label>
      </div>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;
